gtda/externals/python/ripser_interface.py

```diff
@@ -122,6 +122,28 @@ def get_greedy_perm(X, n_perm=None, metric="euclidean"):
     return idx_perm, lambdas, dperm2all
 
 
+def _resolve_symmetry_conflicts(coo):
```

Comment:
Docstrings need adding. This function returns data for an upper triangular COO matrix, starting from a generic COO. If entries (i, j) and (j, i) are both present, the final value at (i, j) is kept the same. If only entry (j, i) is present, the final value at (i, j) is the starting value at (j, i).

Reply:
Just to be sure, what you want to say is that `ripser` only goes for the upper triangle in the distance matrix => j > i, right ?

Reply:
I wanted to explain what this function does. You're asking about why it does it and that has mostly to do with `ripser` I agree (but it affects collapser too). Thanks this function, the user can pass a sparse matrix with some entries in the lower diagonal to the `ripser` function below, and these entries are not ignored. However, if there is already an entry in the transpose position, the conflict is resolved in favour of the upper diagonal entry. The output of this function is an upper-triangular matrix because then we know that ripser C++ won't discard some entries (as you say.

Reply:
Makes sense, thank you.

Reply:
Docstrings and inline comments added

Reply:
Nice ! :+1: